spec/models/tag_spec.rb

```diff
@@ -77,6 +78,81 @@
         expect(@fandom_tag.taggings_count_cache).to eq 40 * ArchiveConfig.TAGGINGS_COUNT_CACHE_DIVISOR
       end
     end
+
+    context "redis" do
+      it "does not write to the database when reading the count" do
+        tag = fandom_tag_with_one_work
+        tag.taggings_count
+        # Check if redis has flagged this tag for an update to the database.
+        expect(REDIS_GENERAL.sismember("tag_update", tag.id)).to eq false
+      end
+
+      it "does not write to the database when assigning the same count" do
+        tag = fandom_tag_with_one_work
+        tag.taggings_count = 1
+        # Check if redis has flagged this tag for an update to the database.
+        expect(REDIS_GENERAL.sismember("tag_update", tag.id)).to eq false
+      end
+
+      it "writes to the database when assigning a new count" do
+        tag = fandom_tag_with_one_work
+        tag.taggings_count = 2
+        # Check if redis has flagged this tag for an update to the database.
+        expect(REDIS_GENERAL.sismember("tag_update", tag.id)).to eq true
+
+        write_tag_to_database(tag)
+        # Actual number of taggings has not changed though count cache has.
+        expect(tag.taggings_count_cache).to eq 2
+        expect(tag.taggings_count).to eq 1
+      end
+
+      it "writes to the database when adding a new work with the same tag" do
+        tag = fandom_tag_with_one_work
+        expect(tag.taggings_count_cache).to eq 1
+        expect(tag.taggings_count).to eq 1
+
+        FactoryBot.create(:work, fandom_string: tag.name)
+        # Check if redis has flagged this tag for an update to the database.
+        expect(REDIS_GENERAL.sismember("tag_update", tag.id)).to eq true
+
+        write_tag_to_database(tag)
+        expect(tag.taggings_count_cache).to eq 2
+        expect(tag.taggings_count).to eq 2
+      end
+
+      it "does not write to the database with a blank value" do
+        tag = fandom_tag_with_one_work
+        # Blank values will cause errors if assigned earlier due to division
+        # in taggings_count_expiry.
+        REDIS_GENERAL.set("tag_update_#{tag.id}_value", "")
+        REDIS_GENERAL.sadd("tag_update", tag.id)
+
+        write_tag_to_database(tag)
+        expect(tag.taggings_count_cache).to eq 1
+      end
+
+      it "triggers reindexing of tags which aren't used much" do
+        tag = fandom_tag_with_one_work
+        FactoryBot.create(:work, fandom_string: tag.name)
+        # Run indexing jobs to prevent automatic reindexing with small tags
+        run_all_indexing_jobs
+
+        expect do
+          write_tag_to_database(tag)
+        end.to(add_to_reindex_queue(tag, :main))
+      end
+
+      it "triggers reindexing of tags which are used significantly" do
+        tag = fandom_tag_with_one_work
+        (1..ArchiveConfig.TAGGINGS_COUNT_MIN_CACHE_COUNT).each do
```

Comment:
I'm not sure this test and the previous one are different. In both cases, you're using tags with use count under `ArchiveConfig.TAGGINGS_COUNT_MIN_CACHE_COUNT` ("small").

Note that in:
https://github.com/otwcode/otwarchive/blob/e7c7da0a33a554cfde124ef1ad2e3df5ce005470/spec/models/tag_spec.rb#L28-L35
It's a large tag because one more work was added outside the loop. The test wants to check the difference when a tag goes from "small" to "large".

If you don't care about that transition, you can simply use `ArchiveConfig.TAGGINGS_COUNT_MIN_CACHE_COUNT.times do` to create enough works for "large".